### PR TITLE
Jl/caip multichain/type cleanups

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

An effort to cleanup some lazy permission types and differentiate the potentially unflattened/malformed ScopeObjects from a CAIP-25 request and the internal flattened ScopeObjects that get persisted into the PermissionController state.

I don't love the current Internal/External prefixing. Perhaps making the external type `unknown` and keeping the unprefixed as the internal type would be better. Thoughts?

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26690?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/lib/multichain-api/scope/scope.ts [hunk 1]
```diff
@@ -16,16 +16,25 @@ export enum KnownCaipNamespace {
...
   rpcDocuments?: string[];
   rpcEndpoints?: string[];
 };
 
-export type ScopesObject = Record<Scope, ScopeObject>;
+export type ExternalScopesObject = Record<Scope, ExternalScopeObject>;
+
+export type InternalScopesObject = Record<CaipChainId, InternalScopeObject> & {
+  [KnownCaipNamespace.Wallet]?: InternalScopeObject;
+};
```
Comment (adonesky1) [inline app/scripts/lib/multichain-api/scope/scope.ts:37]: Could we add a few comments/tsdocs explaining these types?

Comment (adonesky1) [reply app/scripts/lib/multichain-api/scope/scope.ts:37]: Can be pretty lightweight. Just explaining why there is a bifurcation between external and internal scopeObjects

**Review by adonesky1 — APPROVED** [review]
one request for more comments otherwise LGTM!

File: app/scripts/lib/multichain-api/scope/scope.ts (hunk 1 above)
Comment (jiexi) [reply app/scripts/lib/multichain-api/scope/scope.ts]: comments added here https://github.com/MetaMask/metamask-extension/pull/26690/commits/218ad5e1e3c74bc0b69366f09f35e9786a061396

File: app/scripts/lib/multichain-api/scope/scope.ts [hunk 2]
```diff
@@ -49,18 +49,32 @@ export const KnownNotifications: Record<NonWalletKnownCaipNamespace, string[]> =
     eip155: ['accountsChanged', 'chainChanged', 'eth_subscription'],
   };
 
-export type Scope = CaipChainId | CaipReference;
+// These External prefixed types represent the CAIP-217
+// Scope and ScopeObject as defined in the spec.
+export type ExternalScope = CaipChainId | CaipReference;
```
Comment (adonesky1) [inline app/scripts/lib/multichain-api/scope/scope.ts:54]: Ok sorry to switch up again on ya here but wdyt about `ExternalScopeString` and `ScopeString` that's actually [what they're called in CAIP-217](https://github.com/ChainAgnostic/CAIPs/blob/6cf6408c50865b596258f2de476133670aa9aea7/CAIPs/caip-217.md?plain=1#L54) and CAIP-25

**Pull request merged**